core/resource/resource_status.go

```diff
@@ -51,6 +51,15 @@ func (r *Resource) MarkToUpdate() error {
 	return errors.InvalidStateTransition(EntityResource, msg)
 }
 
+func (r *Resource) MarkToDelete() error {
+	if StatusForToDelete(r.status) {
+		r.status = StatusToDelete
+		return nil
+	}
+	msg := fmt.Sprintf("status transition for [%s] from status [%s] to status [%s] is not allowed", r.FullName(), r.status, StatusToUpdate)
```

Comment:
yes, where we use same pattern as create and update in https://github.com/goto/optimus/blob/057436645902a1851aa60bfcf16b92a09e59a3aa/core/resource/service/resource_service.go#L411

so before processing create and update. It will mark them as to create or to update.